src/main/java/org/jetbrains/bsp/bazel/server/bsp/BazelBspServerBuildManager.java

```diff
@@ -0,0 +1,365 @@
+package org.jetbrains.bsp.bazel.server.bsp;
+
+import ch.epfl.scala.bsp4j.BuildTarget;
+import ch.epfl.scala.bsp4j.BuildTargetCapabilities;
+import ch.epfl.scala.bsp4j.BuildTargetDataKind;
+import ch.epfl.scala.bsp4j.BuildTargetIdentifier;
+import ch.epfl.scala.bsp4j.CompileResult;
+import ch.epfl.scala.bsp4j.JvmBuildTarget;
+import ch.epfl.scala.bsp4j.ScalaBuildTarget;
+import ch.epfl.scala.bsp4j.ScalaPlatform;
+import ch.epfl.scala.bsp4j.SourceItem;
+import ch.epfl.scala.bsp4j.SourceItemKind;
+import ch.epfl.scala.bsp4j.WorkspaceBuildTargetsResult;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Lists;
+import com.google.devtools.build.lib.query2.proto.proto2api.Build;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+import org.eclipse.lsp4j.jsonrpc.messages.ResponseError;
+import org.jetbrains.bsp.bazel.common.Constants;
+import org.jetbrains.bsp.bazel.common.Uri;
+import org.jetbrains.bsp.bazel.server.bep.BepServer;
+import org.jetbrains.bsp.bazel.server.data.BazelData;
+import org.jetbrains.bsp.bazel.server.data.ProcessResults;
+import org.jetbrains.bsp.bazel.server.resolver.BazelRunner;
+import org.jetbrains.bsp.bazel.server.resolver.QueryResolver;
+import org.jetbrains.bsp.bazel.server.util.ParsingUtils;
+
+public class BazelBspServerBuildManager {
```

Comment:
This file seems pretty monotholic while still touching on several parts that #23 also focuses on. It's something we should definitely take into consideration

Reply:
Same as [below](https://github.com/JetBrains/bazel-bsp/pull/24#discussion_r549328902).